src/Dhall.hs

```diff
@@ -1333,3 +1340,80 @@ field key valueType =
           )
           ( Data.Functor.Compose.Compose extractBody )
       )
+
+{-| The 'InputRecordType' divisible (contravariant) functor allows you to build 
+    an 'InputType' injector for a Dhall record.
+
+    For example, let's take the following Haskell data type:
+
+> data Project = Project
+>   { projectName :: Text
+>   , projectDescription :: Text
+>   , projectStars :: Natural
+>   }
+
+    And assume that we have the following Dhall record that we would like to
+    parse as a @Project@:
+
+> { name =
+>     "dhall-haskell"
+> , description =
+>     "A configuration language guaranteed to terminate"
+> , stars =
+>     289
+> }
+
+    Our injector has type 'InputType' @Project@, but we can't build that out of any
+    smaller injectors, as 'InputType's cannot be combined (they are only 'Contravariant's).
+    However, we can use an 'InputRecordType' to build an 'InputType' for @Project@:
+
+> injectProject :: InputType Project
+> injectProject =
+>   inputRecord
+>     (  adapt >$< inputField "name" (inject @Text)
```

Comment:
I believe the type applications are not necessary since they should be inferred by the context

Reply:
I think you are right. And actually, given the way the API is written, they should either be `inputFieldWith` or the second parameter should be dropped entirely. The example as written woudn't typecheck!

Reply:
Note that we have `doctest` enabled on this module, so if you transform the example into a doctest then it will be checked 🙂

Reply:
I might do that in a separate PR, as it would involve modifying the $setup block to bring the `Project` type into scope.

Reply:
Yeah, it's not urgent to do in this PR